Fix multiple tests with refactoring driver for duplicate class.

hernanmd opened the pull request:
Fix a couple of tests failing, reported in #16850, due to missing support to pre-configure refactorings for test purposes.
Also configured missing mocks for the preview presenter.
Fix also for #16825

Commit:
Fix multiple tests with refactoring driver for duplicate class.

MarcusDenker commented:
I am doing a revert of the revert: https://github.com/pharo-project/pharo/pull/16854

hernanmd commented:
> I am doing a revert of the revert: #16854

Thanks, I will check further fails and check better next time.
(and I forgot to mention I was working on this issue)

MarcusDenker commented:
Revert fixed the conflicts, tests are now run

MarcusDenker commented on src/Refactoring-UI/ReDuplicateClassDriver.class.st:
```diff
@@ -99,6 +99,17 @@ ReDuplicateClassDriver >> runRefactoring [
 	self applyChanges 
 ]
 
+{ #category : 'as yet unclassified' }
```
this makes testNoUncategorizedMethods fail

MarcusDenker reviewed (changes requested):
this adds one more failing testNoUncategorizedMethods

Commit:
Fix critics

hernanmd committed:
Merge branch 'pharo-project:Pharo13' into fix_rb_tests

Commit:
Fix testThatThereAreNoSelectorsRemainingThatAreSentButNotImplemented in ReClassForGeneratingEqualAndHashExistingImplementors

Commit:
Merge 6c2b6704d5c13669f8c3c50bf22ca6f945f8ee35

MarcusDenker commented on src/Refactoring-Core/RBCopyClassRefactoring.class.st:
```diff
@@ -150,41 +195,38 @@ ReCopyClassRefactoring >> copyVariables [
 
-{ #category : 'accessing' }
-ReCopyClassRefactoring >> instanceMethods: anObject [
+||||||| 3926c07db6:src/Refactoring-Core/ReCopyClassRefactoring.class.st
```
this is now leading to load error

```
MetacelloNotification: Loaded -> Refactoring-Environment --- tonel:///builds/workspace/est_and_branch_Pipeline_PR-16853/64/src()
MetacelloNotification: Loaded -> Refactoring-Changes --- tonel:///builds/workspace/est_and_branch_Pipeline_PR-16853/64/src()
TonelParseError: Can't parse method body
```

Commit:
merge

Commit:
Fix merge conflicts

MarcusDenker commented:
Now load fails:

```
MetacelloNotification: Loaded -> Refactoring-Changes --- tonel:///builds/workspace/est_and_branch_Pipeline_PR-16853/64/src()
Error: Package Refactoring-Core depends on the following classes:
  ReCopyClassRefactoring
You must resolve these dependencies before you will be able to load these definitions: 
  ReCopyClassRefactoring>>#classMethods
  ReCopyClassRefactoring>>#classMethods:
  ReCopyClassRefactoring>>#copyClass
  ReCopyClassRefactoring>>#copyClass:withName:
  ReCopyClassRefactoring>>#copyMethods:of:in:
  ReCopyClassRefactoring>>#instanceMethods

```

Commit:
Fix RB -> Re

MarcusDenker commented:
Now all the test that the revert fixed are failing:

```

ReDuplicateClassDriverTest.testDuplicateClass	
 
 
 osx-64 / Tests-osx-64 / MacOSX64.Refactoring.Transformations.Tests.RBCopyPackageParametrizedTest.testCopyPackageAndChangesCopyReferences(#rbClass->RBCopyPackageRefactoring)	6.8 sec	1
 osx-64 / Tests-osx-64 / MacOSX64.Refactoring.Transformations.Tests.RBCopyPackageParametrizedTest.testCopyPackageWithParameters(#rbClass->RBCopyPackageRefactoring)
```

In addition, we have:

ReleaseTest>>testUndeclared


It might fix other, but honestly, I am starting to get lost completely

Might it not be a better idea to revert ?

guillep commented:
Hi Marcus, I agree with you. We should revert. That was the rule we set.

@hernanmd it is important for us that this gets fixed fast, either by reverting or by fixing it.
It simplifies the management of all the other ~60 PRs we have.

hernanmd commented:
> this adds one more failing testNoUncategorizedMethods

Hi @MarcusDenker, could you share a link to see the failed tests?

MarcusDenker commented:
The failing tests are here:

https://ci.inria.fr/pharo-ci-jenkins2/job/Test%20pending%20pull%20request%20and%20branch%20Pipeline/job/PR-16853/8/testReport/

Commit:
Add rules to ignore critics in TestCase data classes.
Add wrongly deleted methods and instance variables in ReCopyClassRefactoring.
Re-add wrongly deleted SycCmDuplicateClassCommand.

MarcusDenker commented:
Thanks!
